src/abstractdataframe/show.jl

```diff
@@ -157,82 +146,98 @@ function getprintedwidth(maxwidths::Vector{Int}) # -> Int
     return totalwidth
 end
 
-#' @description
-#'
-#' When rendering an AbstractDataFrame to a REPL window in chunks, each of
-#' which will fit within the width of the REPL window, this function will
-#' return the indices of the columns that should be included in each chunk.
-#'
-#' NOTE: The resulting bounds should be interpreted as follows: the
-#'       i-th chunk bound is the index MINUS 1 of the first column in the
-#'       i-th chunk. The (i + 1)-th chunk bound is the EXACT index of the
-#'       last column in the i-th chunk. For example, the bounds [0, 3, 5]
-#'       imply that the first chunk contains columns 1-3 and the second chunk
-#'       contains columns 4-5.
-#'
-#' @param maxwidths::Vector{Int} The maximum width needed to render each
-#'        column of an AbstractDataFrame.
-#' @param splitchunks::Bool Should the output be split into chunks at all or
-#'        should only one chunk be constructed for the entire
-#'        AbstractDataFrame?
-#' @param availablewidth::Int The available width in the REPL.
-#'
-#' @returns chunkbounds::Vector{Int} The bounds of each chunk of columns.
-#'
-#' @examples
-#'
-#' df = DataFrame(A = 1:3, B = ["x", "yy", "z"])
-#' maxwidths = getmaxwidths(df, 1:1, 3:3, "Row")
-#' chunkbounds = getchunkbounds(maxwidths, true)
+"""
+    getchunkbounds(maxwidths::Vector{Int},
+                   availablewidth::Int=displaysize()[2])
+
+When rendering an `AbstractDataFrame`` to a REPL window in chunks, each of
+which will fit within the width of the REPL window, this function will
+return the indices of the columns that should be included in each chunk.
+
+NOTE: The resulting bounds should be interpreted as follows: the
+i-th chunk bound is the index MINUS 1 of the first column in the
+i-th chunk. The (i + 1)-th chunk bound is the EXACT index of the
+last column in the i-th chunk. For example, the bounds [0, 3, 5]
+imply that the first chunk contains columns 1-3 and the second chunk
+contains columns 4-5.
+
+# Arguments
+- `maxwidths::Vector{Int}`: The maximum width needed to render each
+  column of an AbstractDataFrame.
+- `availablewidth::Int`: The available width in the REPL.
+
+# Examples
+```jldoctest
+julia> using DataFrames
+
+julia> df = DataFrame(A = 1:3, B = ["x", "yy", "z"]);
+
+julia> maxwidths = DataFrames.getmaxwidths(df, 1:1, 3:3, :Row)
+3-element Array{Int64,1}:
+ 1
+ 1
+ 3
+
+julia> DataFrames.getchunkbounds(maxwidths)
+2-element Array{Int64,1}:
+ 0
+ 2
+```
+"""
 function getchunkbounds(maxwidths::Vector{Int},
-                        splitchunks::Bool,
                         availablewidth::Int=displaysize()[2]) # -> Vector{Int}
     ncols = length(maxwidths) - 1
     rowmaxwidth = maxwidths[ncols + 1]
-    if splitchunks
```

Comment:
I guess the idea was (before I was working with this code - I just retained it guessing the reason) that with `splitchunks` equal to `false` (which is default for `showall`) be able to dump a whole `DataFrame` to e.g. a file using `showall(io, df)`, where `io` is some stream. Possible usages are (not very common, but possible):

1. someone might want to parse it (`show` with `allcols` set to `true` is inconvenient for this);
2. if you dump `showall` to a file and then look at the file in an editor with line wrapping disabled actually you can inspect a whole `DataFrame`.

If you remove it how can you achieve it?

Reply:
The issue is that `displaysize` for general `IO` falls back to `displaysize()`.

Reply:
Indeed, it's annoying that `displaysize` cannot be used for that. What `Array` does is that either the `IOContext` property `limit` is set, and then only the rows/columns that fit on screen are printed, or it's not and everything is printed. For DataFrames it's a bit more complex because of that chunks feature.

The easiest solution is to add a keyword argument for that. That's not too bad, but I wish we would have been able to integrate more with `IOContext`. There's already a duplication/conflict between `allrows`/`allcols` and the `limit` property. I guess we could fix that by having the latter default to the former.

Reply:
Another option, which I now think is better: repurpose `allcols` so that it implies `splitchunks` when and only when `get(io, :limited, false)` is `true`, and have `allcols=get(io, :limited, false)` by default too. IOW, when the display is limited we only show the first columns by default, but we show all of them in chunks when `allcols=true`. When the display is not limited, print all columns without chunks. That should work well by default both for the REPL (limited) and files (unlimited).

What do you think about this?

Reply:
I was replying to your earlier comment in parallel 😄. Now regarding your proposal. And I guess you wanted to say:

> repurpose allcols so that it implies splitchunks when and only when get(io, :limited, false) is `false`

instead of:

> repurpose allcols so that it implies splitchunks when and only when get(io, :limited, false) is `true`

right?

I think it is clean enough, but it would be good to document the usage of `:limited`. The question about paging when `allrows=true` and there are a lot of rows remains (I am OK with whatever is implemented but I just want to know what we do then).

Reply:
Right.

I don't think we need to do any paging for rows, as scrolling is already good enough. We could improve that later anyway if we want. Horizontal scrolling for columns is an interesting option, but as long as it's not supported by most terminals it should only be enabled when we detect that the terminal supports it.

Reply:
Agreed. So let us go ahead with what you have proposed.

Reply:
I've pushed a commit implementing this. Let me know what you think.